### PR TITLE
feat: stub out to go-libipfs

### PR DESCRIPTION
**Pull request by aschmahmann**
Plan is to rearchive this repo once this is done

### CODE REVIEW

File: version.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-  "version": "v0.2.0"
```
Comment (aschmahmann) [inline version.json:2]: Should this be a patch version bump so people can upgrade and things will work, or minor so they don't have to follow us along to go-libipfs?

**Comment by aschmahmann** [comment]
Any ideas what's going on with `goreleaser`? When I run `go mod tidy` locally everything is fine, and don't I still have the things go-releaser things we're missing?

File: version.json (hunk 1 above)
Comment (Jorropo) [reply version.json:2]: The issue is that `go-libipfs` have newer versions of libs which could break stuff.

For example if I import `example.com@v1.0.0` and go-libipfs imports `example.com@v1.1.0`, updating to `go-ipfs-files@v0.2.1` will update `example.com@v1.0.0` to `example.com@v1.1.0`.

I don't really care about it but that why goreleaser is angry with you when you update go.mod and bump a patch.

File: deprecated.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	// Deprecated: moved to github.com/ipfs/go-libipfs/files
+	ToDir = files.ToDir
+	// Deprecated: moved to github.com/ipfs/go-libipfs/files
+	FileFromEntry = files.FileFromEntry
+	// Deprecated: moved to github.com/ipfs/go-libipfs/files
+	DirFromEntry = files.DirFromEntry
+	// Deprecated: moved to github.com/ipfs/go-libipfs/files
+	Walk = files.Walk
+	// Deprecated: moved to github.com/ipfs/go-libipfs/files
+	NewWebFile = files.NewWebFile
```
Comment (Jorropo) [inline deprecated.go:104]: I usually like to keep the old file structure (\*only files that had public symbols).

This makes the review process much easier.

Comment (aschmahmann) [reply deprecated.go:104]: I separated by commits. You can go back earlier and see. I'm also happy to revert if that's better

File: go.mod [hunk 3]
```diff
@@ -1,15 +1,10 @@
 module github.com/ipfs/go-ipfs-files
 
-require (
-	github.com/crackcomm/go-gitignore v0.0.0-20170627025303-887ab5e44cc3
-	github.com/stretchr/testify v1.7.0
-	golang.org/x/sys v0.0.0-20190302025703-b6889370fb10
-)
+require github.com/ipfs/go-libipfs v0.4.1-0.20230131104030-73465059ab96
```
Comment (aschmahmann) [inline go.mod:3]: TODO: use v0.4.0

Comment (aschmahmann) [reply go.mod]: done

**Comment by hacdias** [comment]
@aschmahmann I think the problem with GoRelease is that functions are now variable. So instead of doing:

```go
NewMultiFileReader = files.NewMultiFileReader
```

Maybe we should:

```go
func NewMultiFileReader(file Directory, form bool) *MultiFileReader {
	return files.NewMultiFileReader(file, form)
}
```

File: version.json (hunk 1 above)
Comment (hacdias) [reply version.json:2]: I updated to 0.3.0

**Review by hacdias — APPROVED** [review]
Reviewed, commit by commit. Thanks for this @aschmahmann! I'm merging and releasing as 0.3.0 for the reasons outlined by @Jorropo. I will archive this repo after the release is complete.

**Pull request merged**